internal/config/client.go

```diff
@@ -1,11 +1,12 @@
-package mongodbatlas
+package config
```

Comment:
@lantoli two questions:

1. can you perhaps highlight in the description your idea of folder structure? Going through each file and try to reverse engineer is long process
2. How does this relate to https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1643 ? Are we moving with both? Is this one intended to cover Agustin proposal and replace that PR? 

Thank you!

Reply:
@marcosuma @AgustinBettati 

1 - I've preferred to describe in the "File Structure" Scope doc so we can comment and discuss about the points.
2 - This PR takes ideas from @AgustinBettati  (and other places) and would replace his PR, both PRs can't be merged. We can compare and choose or even create a third PR with the knowledge we've got :-)

Some differences:
- Moving also the tests with the resources, this is what it's taking more time. Also separate test helper methods from test files. Also have tests in a different package to the code being tested to minimize coupling.
- Minimize the changed files in the beginning so we can proceed smoothly in iterations.
- Using internal as root folder and not having code files in a folder if it has subfolders with code files. That is, there are only code files in leaf folders.
- There are some ideas in the doc that are not so obvios in the PR (because it's only for one resource) like grouping per API tags instead of for each single resource, they could rely apply to any of the PRs.
- Remove redudant parts of the filename. e.g: data_source_mongodbatlas_api_key.go can be just data_source_api_key.go. Also remove fw_.

Reply:
> 1 - I've preferred to describe it in the "File Structure" Spec doc

@lantoli I might have missed it then, where is the Spec doc? Until I see that I find it hard to leave valuable feedback.